pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster/host.go

```diff
@@ -620,6 +620,160 @@ func (h *Host) WaitKubeAPIReady(port int) error {
 	return h.WaitHTTPStatus(fmt.Sprintf("https://localhost:%d/version", port), 200, 401)
 }
 
+// Convert eventTime obtained by kubectl get events to time.Time object.
+func (h *Host) ConvertEventTimeToTimeObject(inputTimestamp string) (time.Time, error) {
+	// Format of eventTime to be obtained with kubectl get events
+	layouts := []string{
+		"2006-01-02T15:04:05Z",
+		"2006-01-02T15:04:05.999999Z",
+	}
+
+	// Convert eventTime to a time object.
+	var t time.Time
+	var err error
+	for _, layout := range layouts {
+		t, err = time.Parse(layout, inputTimestamp)
+		if err == nil {
+			break
+		}
+	}
+
+	return t, err
+}
+
+type KubeSystemEvent struct {
+	Items []struct {
+		Reason    string `json:"reason"`
+		EventTime string `json:"eventTime"`
+	} `json:"items"`
+}
+
+// Returns true if a "schedule" is found from an event newer than the time specified in the argument.
+func (h *Host) FindScheduleFromKubeSystemEvents(now time.Time) (bool, error) {
+	output, err := h.ExecOutput(h.Configurer.K0sCmdf("kubectl -n kube-system get events -o json"), exec.Sudo(h))
```

Comment:
I think something like this should work:

```shell
kubectl -n kube-system get events --field-selector reason=Scheduled --since-time='2023-08-28T00:00:00Z' -o json
```

This should make it possible to remove a lot of extra logic.